### PR TITLE
Run Saucelabs tests for internal PRs except if branch contains 'quick/'

### PR DESCRIPTION
**Pull request by limonte**
Run Saucelabs tests for:

  - internal PRs, except cases when branch contains 'quick/'
  - daily builds, triggered by cron

This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-upload/141)

### CODE REVIEW

File: wct.conf.js [hunk 1]
```diff
@@ -3,8 +3,20 @@ var args = require('yargs').argv;
 module.exports = {
   extraScripts: args.dom === 'shadow' ? ['test/enable-shadow-dom.js'] : [],
   registerHooks: function(context) {
-    if (process.env.SAUCE_BROWSERS) {
-      context.options.plugins.sauce.browsers = JSON.parse(process.env.SAUCE_BROWSERS);
+    // do not run Saucelabs tests if PR branch containt 'quick/'
+    if (process.env.TRAVIS_BRANCH.indexOf('quick/') === -1) {
```
Comment (Saulis) [inline wct.conf.js]: I think we need `process.env.TRAVIS_EVENT_TYPE === 'push' && ...` here to avoid this being triggered for external pull requests?

**Comment by manolo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[wct.conf.js, line 7 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141#-KZL7zid8p76asDjeOT8-r1-7:-KZLCcZRhEwpg4AuJnTX:b-a0v01k) ([raw file](https://github.com/vaadin/vaadin-upload/blob/c240ca15c234c8597cb678d6fa0188f578cd00db/wct.conf.js#L7)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

I think we need `process.env.TRAVIS_EVENT_TYPE === 'push' && ...` here to avoid this being triggered for external pull requests?
</blockquote></details>

The question is whether we want to do quick tests to external PR or always full. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141)*

File: wct.conf.js (hunk 1 above)
Comment (limonte) [inline wct.conf.js]: [Tested](https://travis-ci.org/vaadin/vaadin-upload/builds/185126026), working like expected :+1:

**Comment by manolo** [comment]
Reviewed 2 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[wct.conf.js, line 23 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141#-KZMWSudJdXWAPte6MFJ:-KZMWSueCRggX3NYfSjc:bfkt99y) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb7ae7082c8cd4b29516866da5b4461c981e61de/wct.conf.js#L23)):*
> ```JavaScript
>         'OS X 10.11/iphone@9.2',
>         'OS X 10.11/ipad@9.2',
>         'Linux/android@5.1'
> ```

I still think that it's better to maintain the list of full browsers in the travis.yml.
Any good reason to move it here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141)*

**Comment by limonte** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[wct.conf.js, line 23 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141#-KZMWSudJdXWAPte6MFJ:-KZMZE89P2gZzZsPy2o5:b-pvo5z9) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb7ae7082c8cd4b29516866da5b4461c981e61de/wct.conf.js#L23)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I still think that it's better to maintain the list of full browsers in the travis.yml.
Any good reason to move it here?
</blockquote></details>

This is the JavaScript config file which is more flexible, e.g. we can use conditions.
I have no idea how to implement this case in travis.yml, I mean conditional assigning env variables.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141)*

**Comment by manolo** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[wct.conf.js, line 23 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141#-KZMWSudJdXWAPte6MFJ:-KZ_oWCA734BIY1R0UH5:bwezqa1) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb7ae7082c8cd4b29516866da5b4461c981e61de/wct.conf.js#L23)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

This is the JavaScript config file which is more flexible, e.g. we can use conditions.
I have no idea how to implement this case in travis.yml, I mean conditional assigning env variables.
</blockquote></details>

What I meant is to maintain the same configuration pattern that we had. So we establish this list as an environment variable

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141)*

**Comment by limonte** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[wct.conf.js, line 23 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141#-KZMWSudJdXWAPte6MFJ:-KZ_pMLoG7QokJ3xQWDv:b-pm2yov) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb7ae7082c8cd4b29516866da5b4461c981e61de/wct.conf.js#L23)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

What I meant is to maintain the same configuration pattern that we had. So we establish this list as an environment variable
</blockquote></details>

I see your point and I agree with it, but I don't know how to perform both 

- better maintainability by setting env vars in `.travis.yml`
- ability to use two (or more if needed in future) kinds of builds

at the same time. Any suggestion will be nice.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[wct.conf.js, line 23 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141#-KZMWSudJdXWAPte6MFJ:-K_9K5z_nwAFOlk3xIYa:bc9i1kf) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb7ae7082c8cd4b29516866da5b4461c981e61de/wct.conf.js#L23)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

I see your point and I agree with it, but I don't know how to perform both 

- better maintainability by setting env vars in `.travis.yml`
- ability to use two (or more if needed in future) kinds of builds

at the same time. Any suggestion will be nice.
</blockquote></details>

Like we have in combo-box, using env vars allows to have multiple jobs, doesn't it ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141)*

**Comment by limonte** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[wct.conf.js, line 23 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141#-KZMWSudJdXWAPte6MFJ:-K_9Kdnh8o8Rg-KVxY0T:bu8pzao) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb7ae7082c8cd4b29516866da5b4461c981e61de/wct.conf.js#L23)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Like we have in combo-box, using env vars allows to have multiple jobs, doesn't it ?
</blockquote></details>

combo-box now has the same config as upload or any other component. This is the **new** feature, I submitted it for upload and not for skeleton intentionally - to see the real-world case for quick tests.

If this will be approved and merged, then I'll make the same changes in all components (without creating PRs, direct commits).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141)*

**Comment by manolo** [comment]
:lgtm_strong: 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/141#-:-K_AMNAqSibW7VXJ_OFD:b-uwkowd)*

**Comment by limonte** [comment]
Stupid me :face_with_head_bandage: 

I totally forgot about this PR and committed the same change directly to master after discussing this change in Flowdock and approving the same change for the date-picker https://github.com/vaadin/vaadin-date-picker/pull/318

Closing this PR.

**Pull request closed without merging**